File: Solutions/0303/0303.go
type NumArray struct {
    prefix []int;
}


func Constructor(nums []int) NumArray {
    prefix:=make([]int,len(nums)+1)
    prefix[0]=0
    for i:=0;i<len(nums);i++{
        prefix[i+1]=nums[i]+prefix[i]
    }
    return NumArray{prefix: prefix}
}


func (this *NumArray) SumRange(left int, right int) int {
    return this.prefix[right+1]-this.prefix[left]
}

File: Solutions/0303/0303.js
class NumArray {
  constructor(nums) {
    this.prefix = new Array(nums.length + 1);
    this.prefix[0] = 0;
    for (let i = 0; i < nums.length; ++i)
      this.prefix[i + 1] = nums[i] + this.prefix[i];
  }

  sumRange(left, right) {
    return this.prefix[right + 1] - this.prefix[left];
  }
}
